### PR TITLE
First draft of Accessibility Team DEP

### PR DESCRIPTION
**Pull request by knyghty**

### CODE REVIEW

File: draft/accessibility-team.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+- User-facing parts of Django, such as default HTML output in forms and such,
+  and the django admin and its default theme, HTML, and JavaScript components.
+
+- All websites and other software projects maintained by the DSF, such as the
+  Django website, including documentation, Django Snippets, Django People,
+  the issue tracker, etc.
+
+The responsibilities of the Accessibility Team are expected to change over
+time, and are to be decided by conensus of the team, with input from the
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
time, and are to be decided by consensus of the team, with input from the
```

File: draft/accessibility-team.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+Technical Board as required. to begin, several areas have been identified:
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
Technical Board as required. To begin, several areas have been identified:
```

File: draft/accessibility-team.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+- Deciding on any relevant accessibility guidelines to follow, such as WCAG,
+  and at which conformance level.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst:76]: I can see why it would be valuable to have that discussion, but at the same time it doesn’t seem to me like there needs to be an accessibility team for this to be decided on – from the get-go there was strong consensus on the Groups thread for WCAG2.0 AA, since lots of national laws are based on it – and from there, WCAG2.1 feel like a natural "might as well" (or 2.2, when that becomes a candidate recommendation).

File: draft/accessibility-team.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+- Implementing automated testing to catch simpler issues, working with the ops
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
- Implementing automated testing to catch issues, working with the ops
```

I think "simpler" might not be the right word here, it’s just that not all of WCAG is completely objective and automate-able.

File: draft/accessibility-team.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  team as needed to integrate this into the CI process.
+
+- Fixing accessibility issues in Django and associated projects.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]: I’m not sure if this warrants a separate line but it might be worthwhile to address that we would expect there to be a significant amount of issues initially, and that (for the short term) the team’s role would also be to coordinate that work.

File: draft/accessibility-team.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+   :depth: 3
+   :local:
+
+Abstract
+========
+
+This DEP proposes the formation of an Accessibility Team to encourage projects
+maintained by the Django Software Foundation to be accessible to as diverse a
+group as possible, particularly people with disabilities that make using the
+web more difficult.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst:24]: Would it be worthwhile to link to a resource on who those people are? For example https://www.w3.org/WAI/people-use-web/abilities-barriers/ or https://www.w3.org/WAI/fundamentals/accessibility-usability-inclusion/.

File: draft/accessibility-team.rst [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  accessibility of the project.
+
+To aid in reviewing, the Accessibility Team members shall be added to a team
+in the GitHub organization with read access to relevant repositories, so that
+they may be requested to review pull requests, at the discretion of the author,
+reviewer, or other party.
+
+Many of these duties can be implemented by any contributor, not only by the
+Accessibility Team, however the Accessibility Team exists here as a fallback
+in case nobody else is available to do so.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]: I think the team’s role should be much more than a fallback – there’s clearly coordination work needed that you can’t expect any contributor to pick up on and self-organise to do, as well as needing to have some authority to make decisions that impact the future of Django, like accessibility standards, testing guidelines, and so on.

File: draft/accessibility-team.rst [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+Motivation
+==========
+
+Accessible websites are not technically difficult to create, but accessibility
+is often overlooked when developing features. Django is no exception to this.
+The django admin currently exhibits many accessibility issues, such as forms
+fields missing labels, low color contrast, a lacklustre keyboard-only
+experience among other issues. Problems exist on other sites maintained by the
+DSF, such as the `Django website <https://www.djangoproject.com/>`__,
+`Django Snippets <https://djangosnippets.org/>`_, and others.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst:125]: I think this should also state that there might be problems with Django’s features, for example `as_table` for forms.

File: draft/accessibility-team.rst [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+=========
+
+An alternative is to go on as usual, leaving accessibility in the hands of
+individual contributors when writing and reviewing code. However, given the
+accessibility issues that exist in the Djago admin and on the
+`Django website <https://www.djangoproject.com/>`__, this doesn't seem to be
+enough.
+
+Another option is to implement some basic standards, such as conforming to WCAG
+WCAG and setting up a CI to test. This approach is better than nothing but it
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
and setting up a CI to test. This approach is better than nothing but it
```

**Review by thibaudcolas — COMMENTED** [review]
This looks great to me! Thank you for taking the lead on this @knyghty. I’m not too familiar with Django’s DEPs, so don’t have a good point of comparison, but to me this already feels like more than a good first draft.

A few thoughts about the whole document:

- Overall I think it would be ok for the tone to be stronger. Clearly it’s not enough to go on as usual, currently Django falls quite short when it comes to accessibility, which feels particularly odd considering its general stance on diversity and inclusion.
- I’ve addressed it in separate points but I think overall there is more scope for an accessibility team than this document states. If the accessibility effort was ongoing, had clear compliance targets, tooling, then maybe the team wouldn’t be needed – but kickstarting all of this is a fair amount of work, especially if this ends up covering different properties beyond the Django admin & other framework features. If the team ends up making itself redundant, great, but we’re not there.
- I imagine this will be reviewed by people who might not be too knowledgeable about accessibility, so  it could be useful to have an additional "Resources" section with links to practical things that are useful to know about – whether that’s standards, specific tools, blog posts, etc.

Finally on the "additional responsibilities of the team", here are suggestions:

- Coordinating manual audits – whether the team delivering them directly, or procuring them from others
- Suggesting changes to the contribution methodology (for example testing guidelines for contributors or reviewers)
- Documenting accessibility – Django’s commitment, resources for contributors

File: draft/accessibility-team.rst (hunk 3 above)
Comment (knyghty) [reply draft/accessibility-team.rst:76]: I agree that in principle, this is already decided to some extent. However, I want to leave it in so that the team can decide to follow new standards without e.g. having to go to the mailing list, which for small things like upgrading to a backwards-compatible WCAG version (such as from 2.0 to 2.1) seems unnecessary to me.

File: draft/accessibility-team.rst (hunk 5 above)
Comment (knyghty) [reply draft/accessibility-team.rst]: I'm not sure either. I do think it's important to note that not every issue will be caught in code review, so I do fully expect more issues to crop up over time, even once the initial issues are fixed. Also, there are always improvements to be made that are not necessary per-se, but would still improve the experience. I'll probably reword this slightly.

File: draft/accessibility-team.rst (hunk 6 above)
Comment (knyghty) [reply draft/accessibility-team.rst:24]: 👍  - will add this.

File: draft/accessibility-team.rst (hunk 7 above)
Comment (knyghty) [reply draft/accessibility-team.rst]: Yeah. The intention here is just to make it clear that it's expected that contributors will fix issues once they're in the tracker, and not everything _must_ be done by the a11y team. But I agree this wording isn't great, I'll have a think.

File: draft/accessibility-team.rst (hunk 8 above)
Comment (knyghty) [reply draft/accessibility-team.rst:125]: This is addressed earlier, "User-facing parts of Django, such as default HTML output in forms and such". I'm not fully aware of any a11y issues with it. If you know of any, let me know and I can add them.

**Comment by knyghty** [comment]
@thibaudcolas I've committed all your suggestions and left replies to your specific comments - I'll be making some changes over the coming day or two.

On the tone, I wanted to strike a balance of "we're not doing enough" without apportioning specific blame - I just think it's been overlooked. I think I could make the motivation portion stronger, I will look at revisiting it, it is I think the weakest section.

That's a great point about contribution & documentation too, I will add something about those.

Thanks for the early feedback, it's of huge benefit 👍

File: draft/accessibility-team.rst (hunk 8 above)
Comment (thibaudcolas) [reply draft/accessibility-team.rst:125]: 👍 for `as_table`, it’s simply inappropriate to use tables for layout of form fields like this, or for any kind of layout really. Tables should be for data only.

**Comment by thibaudcolas** [comment]
🙌 let me know if at any point you’d want another round of review!

**Comment by knyghty** [comment]
@thibaudcolas sure, please go ahead, I think I've reworked everything I can immediately think of.

File: draft/accessibility-team.rst [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+Responsibilities
+----------------
+
+The remit of the Accessibility Team shall include all user-facing components
+of projects maintained by the Django Software Foundation. This includes but is
+not limited to:
+
+- User-facing parts of Django, such as default HTML output in forms and such,
+  and the django admin and its default theme, HTML, and JavaScript components.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]: nitpick: could also add "CSS", or reword this to "UI components".

File: draft/accessibility-team.rst (hunk 3 above)
Comment (thibaudcolas) [reply draft/accessibility-team.rst:76]: 👍  good point, I think you’re right.

File: draft/accessibility-team.rst [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+  the accessibility of the project.
+
+To aid in reviewing, the Accessibility Team members shall be added to a team
+in the GitHub organization with read access to relevant repositories, so that
+they may be requested to review pull requests, at the discretion of the author,
+reviewer, or other party.
+
+Many of these duties can be undertaken by any contributor, not only by the
+Accessibility Team, however the Accessibility Team exists to coordinate this
+work and to step in where contributors are not availble and support those who
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
work and to step in where contributors are not available and support those who
```

File: draft/accessibility-team.rst [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+lack the knowledge to do so themselves.
+
+Motivation
+==========
+
+Accessible websites are not technically difficult to create, but accessibility
+is often overlooked when developing features. Django is no exception to this,
+and given Django's commitment to diversity and inclusion, the lack of
+accessibility in many parts of Django falls short of an acceptable standard.
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst:109]: 👏  reads well, critical but not in a negative way.

File: draft/accessibility-team.rst [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+Improving accessibility also has a tendency to improve the experience of other
+users. This is called the
+`curb cut effect <https://alexwlchan.net/2019/01/monki-gras-the-curb-cut-effect/>`_
+While this is mostly noted in the real world, the web also experiences this
+affect. For example, keyboard shortcuts are used to allow keyboard-only users
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
effect. For example, keyboard shortcuts are used to allow keyboard-only users
```

File: draft/accessibility-team.rst [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+maintained by the DSF, such as the
+`Django website <https://www.djangoproject.com/>`__,
+`Django Snippets <https://djangosnippets.org/>`_, and others.
+
+Rationale
+=========
+
+An alternative is to go on as usual, leaving accessibility in the hands of
+individual contributors when writing and reviewing code. However, given the
+accessibility issues that exist in the Djago admin and on the
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
accessibility issues that exist in the Django admin and on the
```

File: draft/accessibility-team.rst [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+`Django website <https://www.djangoproject.com/>`__, this doesn't seem to be
+enough.
+
+Another option is to implement some basic standards, such as conforming to WCAG
+and setting up a CI to test. This approach is better than nothing but it
+lacks a clear process for deciding on these. Ongoing maintenence would also be
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst]:
```suggestion
lacks a clear process for deciding on these. Ongoing maintenance would also be
```

File: draft/accessibility-team.rst [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+
+Resources
+=========
+
+- `Diverse Abilities and Barriers (W3C)
+  <https://www.w3.org/WAI/people-use-web/abilities-barriers/>`_
+- `Accessibility, Usability, and Inclusion (W3C)
+  <https://www.w3.org/WAI/fundamentals/accessibility-usability-inclusion/>`_
+- `Web Content Accessibility Guidelines (WCAG) Overview
+  <https://www.w3.org/WAI/standards-guidelines/wcag/>`_
```
Comment (thibaudcolas) [inline draft/accessibility-team.rst:152]: Could be a good place to mention [ATAG](https://www.w3.org/TR/ATAG20/). Its Part A is "authoring tools should be accessible", which is essentially what we’re discussing here for the Django admin. Its Part B isn’t something we’ve discussed yet, but I’d love to work on that as well if we agreed it was valuable.

**Review by thibaudcolas — APPROVED** [review]
This is all looking great!

File: draft/accessibility-team.rst (hunk 13 above)
Comment (knyghty) [reply draft/accessibility-team.rst]: I'm particularly glad you caught this before embarrassing myself with such mistakes in my _native tongue_ in the deps repo 😛
